checkov/sca_package/output.py

```diff
@@ -0,0 +1,294 @@
+from datetime import datetime, timedelta
+from typing import List, Union, Dict, Any, Tuple
+
+from packaging import version as packaging_version
+from prettytable import PrettyTable, SINGLE_BORDER
+
+from checkov.common.models.enums import CheckResult
+from checkov.common.output.record import Record, DEFAULT_SEVERITY
+from checkov.common.typing import _CheckResult
+
+
+SEVERITY_RANKING = {
+    "critical": 0,
+    "high": 1,
+    "medium": 2,
+    "low": 3,
+    "none": 4,
+}
+
+
+def create_report_record(
+    rootless_file_path: str, file_abs_path: str, check_class: str, vulnerability_details: Dict[str, Any]
+) -> Record:
+    package_name = vulnerability_details["packageName"]
+    package_version = vulnerability_details["packageVersion"]
+    cve_id = vulnerability_details["id"].upper()
+    severity = vulnerability_details.get("severity", DEFAULT_SEVERITY)
+    description = vulnerability_details.get("description")
+    resource = f"{rootless_file_path}.{package_name}"
+
+    check_result: _CheckResult = {
+        "result": CheckResult.FAILED,
+    }
+    code_block = [(0, f"{package_name}: {package_version}")]
+
+    lowest_fixed_version = "N/A"
+    fixed_versions: List[Union[packaging_version.Version, packaging_version.LegacyVersion]] = []
+    status = vulnerability_details.get("status") or "open"
+    if status != "open":
+        fixed_versions = [
+            packaging_version.parse(version.strip()) for version in status.replace("fixed in", "").split(",")
+        ]
+        lowest_fixed_version = str(min(fixed_versions))
+
+    details = {
+        "id": cve_id,
+        "status": status,
+        "severity": severity,
+        "package_name": package_name,
+        "package_version": package_version,
+        "link": vulnerability_details.get("link"),
+        "cvss": vulnerability_details.get("cvss"),
+        "vector": vulnerability_details.get("vector"),
+        "description": description,
+        "risk_factors": vulnerability_details.get("riskFactors"),
+        "published_date": vulnerability_details.get("publishedDate")
+        or (datetime.now() - timedelta(days=vulnerability_details.get("publishedDays", 0))).isoformat(),
+        "lowest_fixed_version": lowest_fixed_version,
+        "fixed_versions": fixed_versions,
+    }
+
+    record = Record(
+        check_id=f"CKV_{cve_id.replace('-', '_')}",
+        bc_check_id=f"BC_{cve_id.replace('-', '_')}",
+        check_name="SCA package scan",
+        check_result=check_result,
+        code_block=code_block,
+        file_path=f"/{rootless_file_path}",
+        file_line_range=[0, 0],
+        resource=resource,
+        check_class=check_class,
+        evaluations=None,
+        file_abs_path=file_abs_path,
+        severity=severity,
+        description=description,
+        short_description=f"{cve_id} - {package_name}: {package_version}",
+        vulnerability_details=details,
+    )
+    return record
+
+
+def calculate_lowest_complaint_version(
```

Comment:
Complaint - something is wrong
Compliant - upholds a certain rule / benchmark :)
```suggestion
def calculate_lowest_compliant_version(
```

Reply:
🙈